proposals/low-level-struct-improvements.md

```diff
@@ -284,403 +177,340 @@ ref struct RS
+| `ref scoped Span<int> s` | *current method* | *current method* | 
+
+The declaration `scoped ref scoped Span<int>` is allowed but is redundant with `ref scoped Span<int>`. The *ref-safe-to-escape* of a value can never exceed the *safe-to-escape* hence once the *safe-to-escape* is restricted via `ref scoped` the *ref-safe-to-escape* is implicitly restricted as well.
+
+This allows for APIs in C# 11 to be annotated such that they have the same rules as C# 10:
 
 ```c#
-ref struct SmallSpan
+Span<int> CreateSpan(scoped ref int parameter)
 {
-    public ref int _field;
+    // Just as with C# 10, the implementation of this method isn't relevant to callers.
+}
 
-    // Notice once again we're back at the same problem as the original 
-    // CreateSpan method: a method returning a ref struct and taking a ref
-    // parameter
-    SmallSpan TrickyRefAssignment(ref int i)
-    {
-        // *safe-to-escape* is outside the current method by current rules.
-        SmallSpan s = default;
+Span<int> BadUseExamples(int parameter)
+{
+    // Legal in C# 10 and legal in C# 11 due to scoped ref
+    return CreateSpan(ref parameter);
 
-        // The *ref-safe-to-escape* of 'i' is the same as the *safe-to-escape*
-        // of 's' hence most assignment rules would allow it.
-        s._field = ref i;
+    // Legal in C# 10 and legal in C# 11 due to scoped ref
+    int local = 42;
+    return CreateSpan(ref local);
 
-        // Error: this must be disallowed for the exact same reasons we can't 
-        // return a Span<T> wrapping the parameter: the consumption rules
-        // believe such state smuggling cannot exist
-        return s;
-    }
+    // Legal in C# 10 and legal in C# 11 due to scoped ref
+    Span<int> span = stackalloc int[42];
+    return CreateSpan(ref span[0]);
+}
+```
 
-    SmallSpan SafeRefAssignment()
-    {
-        int[] array = new int[] { 42, 13 };
-        SmallSpan s = default;
+The `scoped` annotation also means that the `this` parameter of a `struct` can now be defined as `scoped ref T`. Previously it had to be special cased in the rules as `ref` parameter that had different *ref-safe-to-escape* rules than other `ref` parameters (see all the references to including or excluding the receiver in the span safety rules). Now it can be expressed as a general concept throughout the rules which further simplifies them.
 
-        // Okay: the value being assigned here is known to refer to the heap 
-        // hence it is allowed by our rules above because it requires no changes
-        // to existing method invocation rules (hence preserves compat)
-        s._field = ref array[i];
+In addition to parameters the `scoped` annotation can be applied to locals or `struct` instance methods. These annotations have the same impact to lifetimes when applied to locals. For locals these annotations concretely define the lifetimes and override the lifetime that would be inferred via the initializer. 
```

Comment:
"locals *of* `struct` instance methods"?

Reply:
If this actually means that 'scoped' can be applied to the signature of a struct instance method, consider including an example.

Reply:
If this actually means that 'scoped' can be applied to the signature of a struct instance method, consider including an example.